src/std/srfi/srfi-19.scm

```diff
@@ -164,10 +164,10 @@
 
 (define (tm:time-error caller type value)
   (if (member type tm:time-error-types)
-                                     (if value
-                                         (error caller "TIME-ERROR type ~S: ~S" type value)
-                                         (error caller "TIME-ERROR type ~S" type))
-                                     (error caller "TIME-ERROR unsupported error type ~S" type)))
+    (if value
+      (error caller "TIME-ERROR type ~S: ~S" type value)
```

Comment:
These error constructs are wrong; error doesn't do any formatting.
It seems these are leftover from someone's fancy error.

Can you drop the format specifier from the messages? Also, the error message should be first, not the caller.

Reply:
Done.